src/Parlot/Scanner.cs

```diff
@@ -472,5 +472,56 @@ private bool ReadQuotedString(char quoteChar, out TokenResult result)
 
             return true;
         }
+
+        /// <summary>
+        /// Reads a sequence token enclosed in arbritrary start and end characters.
+        /// </summary>
+        /// <remarks>
+        /// This method doesn't escape the string, but only validates its content is syntactically correct.
+        /// The resulting Span contains the original quotes.
+        /// </remarks>
+        public bool ReadNonEscapableSequence(char startSequenceChar, char endSequenceChar, out TokenResult result)
```

Comment:
I don't understand the need of `endSequenceChar`. If the goal of the PR is to read strings that are not escaped, then the method should be called `ReadNonEscapableString` and fetch the first char (delimiter) by itself, like the other ReadString methods are doing in this class.

This one looks more generic, but at the same time I am not convinced about the added complexity of being able to set a different start/end delimiter.

Reply:
the purpose is also to read sql escapes like [ ] in tsql or `` in mysql, ...

Reply:
Cool. Any link to examples of things that are not word and still can be escaped by doubling it?

Reply:
Not sure about the[] example. It's more like a Between parser that exists if there is no escaping to check. Unless we can use [ between[] ? If not then I continue to think no arguments are necessary.

Reply:
I only found a stackoverflow link https://stackoverflow.com/questions/4200351/what-characters-are-valid-in-an-sql-server-database-name, but I can tell you for sure that you can have almost anything as a character as long as you escape it with []. Here is a doc for mysql: https://dev.mysql.com/doc/refman/8.0/en/identifiers.html